sktime/transformations/panel/augmenter.py

```diff
@@ -0,0 +1,1217 @@
+        y : Series or Panel
+            Always ignored, exists for compatibility.
+
+        Returns
+        -------
+        self: a fitted instance of the transformer
+        """
+        if not self._is_fittable:
+            # nothing has to be fit here
+            return self
+        elif self.use_relative_fit and self.relative_fit_type == "fit":
+            # calculate demanded statistical param for each variable over
+            # (a concatenation of) all instances in the given panel.
+            self._n_vars = X.shape[1]  # get number of vars from X
+            self._stats = []
+            for col in range(self._n_vars):  # loop over demanded variables
+                if col not in self.excluded_var_indices:
+                    long_series = pd.Series(dtype="float64")
+                    for row in range(X.shape[0]):  # loop over instances
+                        long_series = long_series.append(
+                            X.iloc[row, col], ignore_index=True
+                        )
+                    self._stats.append(self.relative_fit_stat_fun(long_series))
+                else:
+                    self._stats.append(None)
+            return self
+        # if use_relative_fit is False or relative_fit_type is "fit-transform"
+        # or "instance-wise"
+        else:
+            # in this case no fitting is necessary
+            return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        Parameters
+        ----------
+        X : Series or Panel of pd.DataFrame
+            Uni- or multivariate dataset.
+        y : Series or Panel, optional (default=None)
+            Always ignored, exists for compatibility.
+
+        Returns
+        -------
+        pd.DataFrame: The transformed version of X.
+        """
+        # create copied DataFrame for transformed data
+        Xt = X.copy()
+        # Xt = pd.DataFrame(dtype=object).reindex_like(X).astype(object)
+        self._last_aug_random_variate = (
+            pd.DataFrame(dtype=object).reindex_like(X).astype(object)
+        )
+        # check number of vars
+        if Xt.shape[1] != self._n_vars and self._n_vars is not None:
+            raise ValueError(
+                "The number of variables differs between input "
+                "data (" + str(Xt.shape[1]) + ") and the data "
+                "used for fitting (" + str(self._n_vars) + ")."
+            )
+
+        # fit-transform
+        if (
+            self.use_relative_fit
+            and self.relative_fit_type == "fit-transform"
+            and not self._is_fittable
+        ):
+            # calculate demanded statistical param over (a concatenation of)
+            # all instances for each variable (like in case of "fit" but
+            # directly on the data to be transformed).
+            self._stats = []
+            for col in range(X.shape[1]):  # loop over demanded variables
```

Comment:
Is this block duplicated in `_fit` ?

Reply:
Good point, it seems that we doubled sklearn's `augmenter.fit_transfrom()` here... I don't remember my intention here - I will check. Do you know the exact definition of the `_tag` `fit-in-transform` of the transformer template? Is it whether `fit_transform()` is **generally** allowed or that **only** `fit_transform()` is allowed (or s.th. completely different).

Reply:
`fit-in-transform` says whether `fit` is empty and could, in-principle, be ommitted. An example for that would be the transformer that squares all entries, no fitting needed. A counter-example would be centering, i.e., subtracting the mean on the training set.